src/lib/es5.d.ts

```diff
@@ -1639,7 +1635,7 @@ interface Int8Array {
       * @param thisArg An object to which the this keyword can refer in the callbackfn function.
       * If thisArg is omitted, undefined is used as the this value.
       */
-    map(callbackfn: (value: number, index: number, array: Int8Array) => number, thisArg?: any): Int8Array;
+    map(callbackfn: (value: number, index: number, array: this) => number, thisArg?: any): T;

```

Comment:
Rather than return `T`, either return `this` or return `TypedArray` with a subtyped overload in each specific typed-array constructor (e.g. return `Int32Array` in the `Int32Array` interface).